libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -426,10 +435,56 @@ void AP_Airspeed::read(uint8_t i)
         state[i].healthy = false;
     }
 
-    state[i].last_update_ms = AP_HAL::millis();
+    state[i].last_update_ms = now;
+}
+
+void AP_Airspeed::update_vel_dot(void)
+{
+    // Update and average speed rate of change
+    // Get DCM
+    const Matrix3f &rotMat = AP_AHRS::get_singleton()->get_rotation_body_to_ned();
```

Comment:
`get_singleton` should always be `nullptr` checked. What you want here is actually `AP::ahrs().get_rotation_body_to_ned()`

Reply:
oops

Reply:
Why can't AP::ahrs() dereference a null pointer?

Reply:
Anything that we have a `AP::foo()` for that returns a reference we are promising will always be allocated on vehicles. In practice the other one would always be valid, but the agreement reached is that any call for `get_singleton()` should be nullptr checked, and things we are promising exist then you should use the reference method if you want to skip it. I honestly championed the need for always checking get_singletons as it made the life of scripting better :)

Reply:
I should have asked "when is AP_AHRS instantiated?"
I set a breakpoint in the constructor but gdb couldn't give me a backtrace

Reply:
Ah, it's owned by the vehicle, https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/Plane.h#L235